xbmc/platform/Platform.h

```diff
@@ -65,4 +94,43 @@ class CPlatform
    * Logs platform specific system info during application creation startup
    */
   virtual void PlatformSyslog() {}
+
+  /**\brief Get a platform service instance.
+   */
+  template<class T>
+  std::shared_ptr<T> GetService()
+  {
+    std::unique_lock<CCriticalSection> lock(m_critSection);
+    const auto it = m_services.find(std::type_index(typeid(T)));
+    if (it != m_services.end())
+      return std::static_pointer_cast<T>((*it).second);
+
+    return nullptr;
+  }
+
+protected:
+  /**\brief Register a new platform service instance.
+   */
+  void RegisterService(const std::shared_ptr<IPlatformService>& service)
+  {
+    if (!service.get())
```

Comment:
shared_ptr has an operator(bool) that doesn't require exposing pointers: `if (!service)`

Reply:
Thanks.